mephisto/abstractions/blueprints/mixins/screen_task_required.py

```diff
@@ -0,0 +1,202 @@
+    )
+
+
+class ScreenTaskRequired(BlueprintMixin):
+    """
+    Compositional class for blueprints that may have a first task to
+    qualify workers who have never attempted the task before
+    """
+
+    def init_mixin_config(
+        self,
+        task_run: "TaskRun",
+        args: "DictConfig",
+        shared_state: "ScreenTaskSharedState",
+    ) -> None:
+        return self.init_screening_config(task_run, args, shared_state)
+
+    def init_screening_config(
+        self,
+        task_run: "TaskRun",
+        args: "DictConfig",
+        shared_state: "ScreenTaskSharedState",
+    ) -> None:
+        self.use_screening_task = args.blueprint.get("use_screening_task", False)
+        if not self.use_screening_task:
+            return
+
+        # Runs that are using a qualification task should be able to assign
+        # a specially generated unit to unqualified workers
+        self.passed_qualification_name = args.blueprint.passed_qualification_name
+        self.failed_qualification_name = args.blueprint.block_qualification
+        self.generate_screening_unit_data: Tuple[
+            bool, ScreenUnitDataGenerator
+        ] = shared_state.generate_screening_unit_data
+        self.screening_units_launched = 0
+        self.screening_unit_cap = args.blueprint.max_screening_units
+
+        find_or_create_qualification(task_run.db, self.passed_qualification_name)
+        find_or_create_qualification(task_run.db, self.failed_qualification_name)
+
+    @classmethod
+    def assert_task_args(cls, args: "DictConfig", shared_state: "SharedTaskState"):
+        use_screening_task = args.blueprint.get("use_screening_task", False)
+        if not use_screening_task:
+            return
+        passed_qualification_name = args.blueprint.passed_qualification_name
+        failed_qualification_name = args.blueprint.block_qualification
+        assert args.task.allowed_concurrent == 1, (
+            "Can only run this task type with one allowed task at a time per worker, to ensure screening "
+            "before moving into more tasks."
+        )
+        assert (
+            passed_qualification_name is not None
+        ), "Must supply an passed_qualification_name in Hydra args to use a qualification task"
+        assert (
+            failed_qualification_name is not None
+        ), "Must supply an block_qualification in Hydra args to use a qualification task"
+        assert hasattr(shared_state, "generate_screening_unit_data"), (
+            "You must supply a generate_screening_unit_data function in your SharedTaskState to use "
+            "qualification tasks."
+        )
+        max_screening_units = args.blueprint.max_screening_units
+        assert max_screening_units is not None, (
+            "You must supply a blueprint.max_screening_units argument to set the maximum number of "
+            "additional tasks you will pay out for the purpose of validating new workers. "
```

Comment:
I would rephrase "validating new workers." => "screening new workers." here as well to keep terminology consistent.

This would also be a good place to remind workers of the charges associated.

We can probably add to this to the end, similar to the README: "Note that you **do pay** for screening units, they are just like any other units."

Unrelated: but I see units and tasks being used quite interchangeable in the docs.

Reply:
Yes, I need to bring the terminology to units across everywhere that makes sense! Will also catch the remaining validation name changes